File: cs/XDust/XDust/XDustInlinePartialNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public class XDustInlinePartialNode : XDustNodeList, IXDustSectionNode
    {
        public XDustInlinePartialNode(String name)
            : base(null)
        {
            this.Name = name;
            this.Parameters = new Dictionary<string, XDustNode>();
        }

        public String Name { get; set; }
        public Dictionary<String, XDustNode> Parameters { get; private set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("{<").Append(this.Name).Append("}");
            sb.Append(base.ToString());
            sb.Append("{/").Append(this.Name).Append("}");
            return sb.ToString();
        }

        public override XDustNode AsBlock()
        {
            return new XDustNodeList(this.Nodes);
        }

        public override string Render(XDust dust, RenderChain chain, Context context, object model)
        {
            return String.Empty;
        }

        #region IXDustSectionNode Members


        public XDustNodeList StartBody(string name)
        {
            throw new NotImplementedException();
        }

        public void EndBody()
        {
            throw new NotImplementedException();
        }

        #endregion
    }
}

File: java/src/xdust/src/com/heydanno/xdust/XDustInlinePartialNode.java
package com.heydanno.xdust;

import java.util.HashMap;
import java.util.Map;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;

public class XDustInlinePartialNode extends XDustNodeList implements
		IXDustSectionNode {

	public XDustInlinePartialNode(String name) {
		super(null);
		this.setName(name);
	}

	private String name;
	private Map<String, XDustNode> parameters;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	@Override
	public Map<String, XDustNode> getParameters() {
		if (null == this.parameters) {
			this.parameters = new HashMap<String, XDustNode>();
		}
		return this.parameters;
	}

	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append("{<").append(this.getName()).append("}");
		sb.append(super.toString());
		sb.append("{/").append(this.getName()).append("}");
		return sb.toString();
	}

	@Override
	public XDustNode asBlock() {
		return new XDustNodeList(this.getNodes());
	}

	@Override
	public String render(XDust dust, RenderChain chain, Context context,
			Object model) throws Exception {
		return "";
	}

	@Override
	public XDustNodeList startBody(String name) {
		throw new NotImplementedException();
	}

	@Override
	public void endBody() {
		throw new NotImplementedException();
	}

}